frontend/pages/clubrank.tsx

```diff
@@ -0,0 +1,134 @@
+import {
+  Container,
+  Icon,
+  InfoPageTitle,
+  Metadata,
+  StrongText,
+  Text,
+} from 'components/common'
+import equal from 'deep-equal'
+import { ReactElement, useEffect, useRef, useState } from 'react'
+import { PaginatedClubPage, renderListPage } from 'renderPage'
+
+import RankDropdown from '~/components/ClubRank/RankDropdown'
+import PaginatedClubDisplay from '~/components/PaginatedClubDisplay'
+import { SearchInput } from '~/components/SearchBar'
+import { SNOW } from '~/constants/colors'
+import { doApiRequest } from '~/utils'
+
+import { ListLoadIndicator, SplashProps } from '.'
+
+const TopSearchBar = ({ onChange }): ReactElement => {
+  const searchTimeout = useRef<number | null>(null)
+  const [searchValue, setSearchValue] = useState<string>('')
+
+  const updateSearchValue = (value: string): void => {
+    if (searchTimeout.current != null) {
+      window.clearTimeout(searchTimeout.current)
+    }
+    setSearchValue(value)
+    searchTimeout.current = window.setTimeout(() => {
+      searchTimeout.current = null
+      onChange(value)
+    }, 100)
+  }
+
+  return (
+    <div className="control has-icons-right">
+      <input
+        className="input"
+        placeholder={`Search for clubs`}
+        value={searchValue}
+        onChange={(e) => updateSearchValue(e.target.value)}
+      />
+      {searchValue.length > 0 && (
+        <span
+          className="icon is-small is-right"
+          onClick={() => {
+            setSearchValue('')
+            onChange('')
+          }}
+          style={{ pointerEvents: 'auto', cursor: 'pointer' }}
+        >
+          <Icon name="x" />
+        </span>
+      )}
+    </div>
+  )
+}
+
+function Rank(props: SplashProps): ReactElement {
+  const [isLoading, setLoading] = useState<boolean>(false)
+  const [searchInput, setSearchInput] = useState<SearchInput>({})
+  const [clubs, setClubs] = useState<PaginatedClubPage>(props.clubs)
+  const currentSearch = useRef<SearchInput>({})
+  useEffect((): void => {
+    const performSearch = async (): Promise<void> => {
+      if (equal(searchInput, currentSearch.current)) {
+        return
+      }
+      setLoading(true) // Only show loading animation if actively searching, rather than refreshing.
+      await search()
+      setLoading(false)
+    }
+
+    performSearch()
```

Comment:
consider using IIFC,

```typescript
useEffect(() => {
    (async () => { ... })()
}, [...])
```

Reply:
Thank you for making me a better person than I am today.